### ISSUE TITLE
We're incorrectly pluralizing table names in some non-English languages

### ISSUE THREAD

**Issue by mazameli** [metabase/metabase#28559]
In an attempt to make the UI look and feel natural, we sometimes pluralize the names of tables, like in this example. Note that I manually changed the name of the `Orders` table to `Order` to show how the drill-through menu here pluralizes it to `Orders`:

<img width="358" alt="image" src="https://user-images.githubusercontent.com/2223916/220797749-50314013-02d6-482a-9881-a798b7df8975.png">

The problem is that all we're doing is naively appending an `s`, which leads to grammatically incorrect situations like this. In Italian, the singular for "order" is "Ordine" (which is what I manually changed the table name to in the below example), but the plural is "ordini," not "ordines." So the text in the popover here is wrong:

<img width="414" alt="image" src="https://user-images.githubusercontent.com/2223916/220798156-b197a674-a115-4686-a992-64f961aecdd7.png">

**Comment by brunobergher** [comment]
If don't already, we may benefit from a library like [numerous](https://www.npmjs.com/package/numerous)

**Comment by rafpaf** [comment]
Can reproduce.

Unfortunately, the `numerous` npm library is not truly capable of pluralizing arbitrary nouns across languages. You have to supply, as input to that library's main function, "an object that holds all possible pluralization forms of the word you want to pluralize" ([source](https://gist.github.com/slavafomin/f2e5259cab17d55af5d9fa4c2c2baa08)). Pluralization across languages is a computationally difficult problem since you have to deal with the irregular plurals in each language. Even regular plurals depend on a variety of factors like case, gender, etc.

We might have to get clever and somehow avoid using the plural form of the noun.

For inspiration, here's an old joke about [mongeese](https://www.reddit.com/r/Jokes/comments/2zufjl/i_was_thinking_of_starting_up_a_small_zoo_so_i/).

**Comment by noahmoss** [comment]
@rafpaf Yeah, we have the ability to translate strings that require pluralization on both the frontend (with ttag) and the backend ([hand-written by yours truly](https://github.com/metabase/metabase/blob/master/src/metabase/util/i18n/plural.clj) 😆 )

But this only works for strings which are known ahead of time and included in our i18n bundle, so that translators can translate them into each possible pluralization form. It's a different story for table names which are dynamic and provided by users. I'm not sure if there's a good way to do that.

**Comment by noahmoss** [comment]
I think a middle ground could be to just define a hard-coded list of common table names that we'd want to pluralize and run them through our normal i18n process. And don't try to pluralize anything that's not in that list (maybe have some kind of fallback that would still be grammatical, like `View this User's <Table name> rows`).

We'd just need to figure out a list that captures all the common table types, but that's not so huge that it makes it onerous for the translators.

**Comment by taqattack** [comment]
This happens for non-English words using English localization settings. For example, word the word "reparaties" get changed to "reparaty" which should techincally be "reparatie"

**Comment by ranquild** [comment]
We use `inflection` for singulazation/pluralization. This is the full list of words it supports. For everything else, it's just about adding `s` in the end:

```
[ regex.singular.man     ],
    [ regex.singular.person  ],
    [ regex.singular.child   ],
    [ regex.singular.ox      ],
    [ regex.singular.axis    ],
    [ regex.singular.octopus ],
    [ regex.singular.alias   ],
    [ regex.singular.summons ],
    [ regex.singular.bus     ],
    [ regex.singular.buffalo ],
    [ regex.singular.tium    ],
    [ regex.singular.sis     ],
    [ regex.singular.ffe     ],
    [ regex.singular.hive    ],
    [ regex.singular.aeiouyy ],
    [ regex.singular.x       ],
    [ regex.singular.matrix  ],
    [ regex.singular.mouse   ],
    [ regex.singular.foot    ],
    [ regex.singular.tooth   ],
    [ regex.singular.goose   ],
    [ regex.singular.quiz    ],
    [ regex.singular.whereas ],
    [ regex.singular.criterion ],
    [ regex.singular.genus ],
```

I think we should re-work the message here so what we don't use this library for it.

**Comment by AlessioLaiso** [comment]
I'm trying to imagine "See these Orders records" in Italian and it still wouldn't work, really. Can we just use the generic "See these records", at least for non-English or specific languages?

**Comment by rafpaf** [comment]
To my ears, "See these Orders records" doesn't sound good in English if you're looking at it out of context. But when you're looking at a question called Orders, which clearly has records, "Orders records" makes enough sense. I wonder if it's the same in Italian.<br><br>If nothing of great value is lost by switching to "See these records", that might be the best approach.
